core/src/main/java/org/jdbi/v3/core/mapper/reflect/BeanMapper.java

```diff
@@ -120,158 +80,18 @@ public static RowMapperFactory factory(Class<?> type, String prefix) {
         return new BeanMapper<>(type, prefix);
     }
 
-    private final Class<T> type;
-    private final String prefix;
-    private final BeanInfo info;
-    private final Map<PropertyDescriptor, BeanMapper<?>> nestedMappers = new ConcurrentHashMap<>();
-
-    private BeanMapper(Class<T> type, String prefix) {
-        this.type = type;
-        this.prefix = prefix.toLowerCase();
-        try {
-            info = Introspector.getBeanInfo(type);
-        } catch (IntrospectionException e) {
-            throw new IllegalArgumentException(e);
-        }
+    private BeanMapper(Class<T> type, PojoProperties<T> properties, String prefix) {
+        super(type, properties, prefix);
+        strictColumnMapping = false;
```

Comment:
No, this is orthogonal, but with a confusingly similar name.  This is declaring that we must find a column mapper for all property types we encounter.  In the original BeanMapper, if we can't figure out how to handle a type, we just assume it's a vendor type and call `rs.getObject`: https://github.com/jdbi/jdbi/blob/master/core/src/main/java/org/jdbi/v3/core/mapper/reflect/BeanMapper.java#L192-L193

I think this behavior is a shitty v2 holdover, personally, so the new code opts to instead throw an exception rather than fallback to `getObject` and hope.

The boolean allows the new implementation to behave one way and the existing `BeanMapper` another.

Reply:
I'll change it to `strictColumnTypeMapping` how's that?